### DISCUSSION TITLE
Calling specific in-built objective function

### DISCUSSION THREAD

**Discussion by Sukantabasu** [microsoft/FLAML#1260] [Q&A]
Hi, 

FLAML provides several in-built "metrics". However, I could not find an easy way to change the "objectives". For example, lightgbm provides several objectives (https://lightgbm.readthedocs.io/en/latest/Parameters.html). For regression problems, one can use rmse, mae, Huber, etc. Is there a way I can specify one of these pre-existing objectives? I am not referring to any custom objectives. 

Thanks.

**Comment by sonichi** [comment]
Depending on whether the parameter is set in constructor or .fit(), use either
https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#a-shortcut-to-override-the-search-space
or 
https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#extra-fit-arguments

**Reply by Sukantabasu** [reply]
Thanks a lot!

**Reply by Sukantabasu** [reply]
If I am not mistaken, "objective" should be defined via "custom_hp" option. However, I cannot figure out what should be the "domain".

**Reply by sonichi** [reply]
The value you want to use if you want to fix it.

**Reply by Sukantabasu** [reply]
I want to use "mae". So, the domain is "mae"?

**Reply by Sukantabasu** [reply]
If I try: 
custom_hp = {
    "lgbm": {
        "objective": 
            {"domain": "mae",  
            }
    },
}

I am getting the error: 
1894             "period"
   1895         ),  # NOTE: this is after kwargs is updated to fit_kwargs_by_estimator
-> 1896         custom_hp=custom_hp and custom_hp.get(estimator_name),
   1897         max_iter=max_iter / len(estimator_list) if self._learner_selector == "roundrobin" else max_iter,
   1898         budget=self._state.time_budget,
   1899     )
   1900 logger.info("List of ML learners in AutoML Run: {}".format(estimator_list))
   1901 self.estimator_list = estimator_list

AttributeError: 'str' object has no attribute 'get'
